client/webserver/middleware.go

```diff
@@ -86,3 +98,33 @@ func (s *WebServer) requireDEXConnection(next http.Handler) http.Handler {
 		next.ServeHTTP(w, r)
 	})
 }
+
+// orderIDCtx embeds order ID into the request context
+func orderIDCtx(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		oid := chi.URLParam(r, "oid")
+		ctx := context.WithValue(r.Context(), ctxOID, oid)
+		next.ServeHTTP(w, r.WithContext(ctx))
+	})
+}
+
+func getOrderIDCtx(r *http.Request) (dex.Bytes, error) {
+	oidStr, ok := r.Context().Value(ctxOID).(string)
+	if !ok {
+		log.Errorf("getOrderIDCtx type assertion failed. Expected string, got %T", r.Context().Value(ctxOID))
+		return nil, fmt.Errorf("type assertion failed")
+	}
+	if len(oidStr) != order.OrderIDSize*2 {
+		log.Errorf("getOrderIDCtx received order ID string of wrong length. wanted %d, got %d",
+			order.OrderIDSize*2, len(oidStr))
+		return nil, fmt.Errorf("invalid order ID")
+	}
+	oidB, err := hex.DecodeString(oidStr)
+	if err != nil {
+		log.Errorf("getOrderIDCtx received invalid hex for order ID %q", oidStr)
+		return nil, fmt.Errorf("")
+	}
+	var oid dex.Bytes
+	copy(oid[:], oidB[:])
+	return oid, nil
```

Comment:
`return oidB` without the `copy`.  Also looks like the `copy` wasn't actually copying because `len(oid)==0`.

Reply:
Did a quick go test like:

```
func randOrderID() (oid order.OrderID) {
	rand.Read(oid[:])
	return
}

func Test_getOrderIDCtx(t *testing.T) {
	oid := randOrderID()
	r := &http.Request{}
	r = r.WithContext(context.WithValue(context.Background(), ctxOID, oid.String()))

	bytesOut, err := getOrderIDCtx(r)
	t.Log(bytesOut, err)
}
```

Changing `ctxOID` above to 123 demonstrates the panic.

Reply:
Now I'm wondering how this ever worked for me at all. Hmm. Anyway, fixed now.